timetypes/duration_type.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package timetypes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+var (
+	_ basetypes.StringTypable = (*DurationType)(nil)
+)
+
+// DurationType is an attribute type that represents a valid Go time duration string.
+// See https://pkg.go.dev/time#ParseDuration for more details
+type DurationType struct {
```

Comment:
Given that the Go standard library `time` package is just one potential time duration implementation, we should explicitly define the type and value type to differentiate it from other potential duration implementations in the future. For example, [ISO 8601 implements its own duration format](https://en.wikipedia.org/wiki/ISO_8601#Durations).

```suggestion
type GoDurationType struct {
```

This change will need to be applied elsewhere, such as method signatures and file naming.

Reply:
Done :heavy_check_mark: